File: Assignment10/program4.c
////////////////////////////////////////////////////////////////
//
//  File Name   : program4.c
//  Description : Accept a Integer and calculates multiplication of Digits
//  Author      : Akhilesh.P.Sonavane.
//  Date        : 17/05/2025
//
////////////////////////////////////////////////////////////////

#include<stdio.h>

////////////////////////////////////////////////////////////////
// 
//  Function Name : MultDigits
//  Description   : Returns Multiplication of all digit in input integer
//  Input         : Integer
//  Output        : Integer
//
////////////////////////////////////////////////////////////////

int MultDigits(int iNo)
{
    int iDigit = 0;
    int iFreq = 1;
    
    if(iNo < 0)
    {
        iNo = -iNo;
    }
    
    while (iNo != 0)
    {
        iDigit = iNo % 10;
        
        if(iDigit == 0)
        {
            iDigit = 1;
        }

        iFreq  = iFreq * iDigit;
        iNo = iNo / 10;
    }
    return iFreq;
}

int main()
{
    int iValue = 0;
    int iRet = 0;
 
    printf("enter number");
    scanf("%d",&iValue);

    iRet=MultDigits(iValue);
 
    printf("%d",iRet);
 
    return 0;
}

File: Assignment21/program4.java
////////////////////////////////////////////////////////////////
//
//  File Name   : program4.java
//  Description : Accept a N Integer & displays all three digit Values  
//  Author      : Akhilesh.P.Sonavane.
//  Date        : 29/05/2025
//
////////////////////////////////////////////////////////////////

import java.util.Scanner;

////////////////////////////////////////////////////////////////
// 
//  Class Name    : Display
//  Function Name : Digit
//  Description   : Inputs N Integer & displays all three digit Values 
//  Input         : Integer
//  Output        : Integer
//
////////////////////////////////////////////////////////////////

class Display
{
    public void Digit(int[] iPtr, int iSize)
    {
        int iCnt = 0;

        for(iCnt = 0; iCnt < iSize; iCnt++)
        {
            if((iPtr[iCnt] >= 100) && iPtr[iCnt] <= 999)
            {
                System.out.println("Value "+(iCnt+1)+" : "+iPtr[iCnt]);
            }
        }
    }
}
public class program4
{
    public static void main(String Args[])
    {
        int iCnt = 0;
        int[] iArr = null;
        int iLength = 0;
    
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter Number of Elements : ");
        iLength = sobj.nextInt();

        iArr = new int[iLength];

        for(iCnt = 0; iCnt < iLength; iCnt++)
        {
            System.out.println("Element "+(iCnt+1)+" : ");
            iArr[iCnt] = sobj.nextInt();
        }

        Display dobj = new Display();

        dobj.Digit(iArr, iLength);

        sobj.close();
    }
}

